subprojects/core/src/test/groovy/org/gradle/api/tasks/SourceTaskTest.groovy

```diff
@@ -40,5 +40,33 @@ class SourceTaskTest extends AbstractTaskTest {
         expect:
         task.source.asList() == [file1, file2]
     }
+
+    def "can prepend to source"() {
```

Comment:
Great tests, but I'm unsure about the change where we explicitly check for `UnionFileCollection`.